contracts/FleekERC721.sol

```diff
@@ -267,6 +291,164 @@ contract FleekERC721 is ERC721, FleekAccessControl {
+     *
+     * May emit a {NewMirror} event.
+     *
+     * Requirements:
+     *
+     * - the tokenId must be minted and valid.
+     *
+     * IMPORTANT: The payment is not set yet
+     */
+    function addMirror(uint256 tokenId, string memory mirrorName) public payable {
+        // require(msg.value == 0.1 ether, "You need to pay at least 0.1 ETH"); // TODO: define a minimum price
+        _requireMinted(tokenId);
+        require(_mirrors[mirrorName].owner == address(0), "Mirror already exists");
+
+        _mirrors[mirrorName] = Mirror(tokenId, _apps[tokenId].mirrors.length, 0, msg.sender);
+        _apps[tokenId].mirrors.push(mirrorName);
+
+        emit NewMirror(mirrorName, tokenId, msg.sender);
+    }
+
+    /**
+     * @dev Remove a mirror registry for an app token.
+     * Only the owner of the mirror can remove it.
+     * It will also remove the mirror from the app token mirrors list.
+     *
+     * May emit a {RemovedMirror} event.
+     *
+     * Requirements:
+     *
+     * - the mirror must exist.
+     */
+    function removeMirror(string memory mirrorName) public requireMirror(mirrorName) {
+        require(msg.sender == _mirrors[mirrorName].owner, "You are not the owner of this mirror");
+        App storage _app = _apps[_mirrors[mirrorName].tokenId];
+
+        // the index of the mirror to remove
+        uint256 indexToRemove = _mirrors[mirrorName].index;
+
+        // the last item is reposited in the index to remove
+        string memory lastMirror = _app.mirrors[_app.mirrors.length - 1];
+        _app.mirrors[indexToRemove] = lastMirror;
+        _mirrors[lastMirror].index = indexToRemove;
+
+        // remove the last item
+        _app.mirrors.pop();
+
+        delete _mirrors[mirrorName];
+        emit RemovedMirror(mirrorName, msg.sender);
+    }
+
+    /**
+     * @dev A view function to gether information about a mirror.
+     * It returns a JSON string representing the mirror information.
+     *
+     * Requirements:
+     *
+     * - the mirror must exist.
+     *
+     */
+    function mirror(string memory mirrorName) public view requireMirror(mirrorName) returns (string memory) {
+        Mirror storage _mirror = _mirrors[mirrorName];
+
+        // prettier-ignore
+        bytes memory mirrorJSON = abi.encodePacked(
+            "{",
+                '"tokenId":', Strings.toString(_mirror.tokenId), ",",
+                '"score":', Strings.toString(_mirror.score), ",",
+                '"owner":"', Strings.toHexString(uint160(_mirror.owner), 20), '"',
+            "}"
+        );
+
+        return string(mirrorJSON);
+    }
+
+    /**
+     * @dev A view function to check if a mirror is verified.
+     * A mirror is verified if its score is greater than 0.
+     *
+     * Requirements:
+     *
+     * - the mirror must exist.
+     *
+     */
+    function isMirrorVerified(string memory mirrorName) public view requireMirror(mirrorName) returns (bool) {
+        return _mirrors[mirrorName].score > 0;
```

Comment:
Let's keep verification separate from the score as that can become confusing. In cases where a mirror is delivering the right content and should be verified but has become a subject to intense censorship, the score because of inaccessibility can drop to 0 and that would falsely identify the mirror as unverified.